### PR TITLE
Adding AdvancedCronJob api as per proposal

### PR DESCRIPTION
**Pull request by rishi-anand**
### Ⅰ. Describe what this PR does
This PR creates api objects based on proposal in PR https://github.com/openkruise/kruise/pull/437

### Ⅱ. Does this pull request fix one issue?

#215 

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
Will add test cases in final PR

### Ⅴ. Special notes for reviews
@Fei-Guo, Please review api changes.

### CODE REVIEW

File: controllers/apps/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*
```
[the remark this answers is not in the document]
Comment (rishi-anand) [reply controllers/apps/suite_test.go:1]: @Fei-Guo, Controller will be moved to `pkg/controller/advancedcronjob` folder. This PR is mainly for api and generated codes.

File: docs/proposals/20201024-advancedcronjob.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
```
[the remark this answers is not in the document]
Comment (rishi-anand) [reply docs/proposals/20201024-advancedcronjob.md]: Done

**Comment by rishi-anand** [comment]
@Fei-Guo, Made controller changes and moved it to `pkg/controller` folder. Will use `./test` for e2e tests. Removed `go-logr` dependency from vendor.

**Review by Fei-Guo — COMMENTED** [review]
LGTM
 
Thanks for addressing the comments.

File: pkg/controller/advancedcronjob/advancedcronjob_controller.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+		klog.Error(err)
+		return err
+	}
+
+	return nil
+}
+
+type realClock struct{}
+
+func (_ realClock) Now() time.Time { return time.Now() }
```
Comment (Fei-Guo) [inline pkg/controller/advancedcronjob/advancedcronjob_controller.go:111]: This seems to cause CI failure.

Comment (rishi-anand) [reply pkg/controller/advancedcronjob/advancedcronjob_controller.go]: Fixed build failure.

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**